experimenter/experimenter/experiments/models.py

```diff
@@ -1258,6 +1258,65 @@ def timeline(self):
 
         return timeline_entries
 
+    @property
+    def default_metrics(self):
+        analysis_data = self.results_data.get("v3", {}) if self.results_data else {}
+        other_metrics = analysis_data.get("other_metrics", {})
+        metrics_metadata = analysis_data.get("metadata", {}).get("metrics", {})
+        default_metrics = {}
+
+        for value in other_metrics.values():
+            for metricKey, metricValue in value.items():
+                default_metrics[metricKey] = metrics_metadata.get(metricKey, {}).get(
+                    "friendlyName", metricValue
+                )
+
+        return default_metrics
+
+    def get_branch_data(self, analysis_basis, selected_segment):
+        overall_results = (
+            (
+                self.results_data.get("v3", {})
+                .get("overall", {})
+                .get(analysis_basis, {})
+                .get(selected_segment, {})
+            )
+            if self.results_data
+            else {}
+        )
+
+        branch_data = []
+        branches = []
+
+        if self.reference_branch:
+            branches.append(self.reference_branch)
+        branches.extend(self.treatment_branches)
+
+        for branch in branches:
```

Comment:
branch iteration order doesn't seem to matter here so we can:


```suggestion
        for branch in self.branches.all():
```

We should probably prefetch the screenshot FK relationship:

```
self.branches.all().prefetch_related("screenshots")
```

otherwise we do N + 1 queries instead of 2